### PR TITLE
Allow for many attributes instead of just one

### PR DESCRIPTION
**Pull request by millerts**
Open exam needs to find a couple attributes when using a non guest student.  Adding the ability to find multiple attributes

### CODE REVIEW

File: service/src/test/java/tds/student/services/impl/RtsServiceImplTest.java [hunk 1]
```diff
@@ -78,7 +79,7 @@ public void shouldReturnEmptyIfAttributeCouldNotBeFoundInPackage() throws Except
         when(rtsRepository.findRtsStudentPackage("client", 1)).thenReturn(Optional.of(blob));
         when(packageReader.read(blob)).thenReturn(true);
 
-        Optional<RtsStudentPackageAttribute> maybeAttribute = rtsService.findRtsStudentPackageAttribute("client", 1, "attribute");
-        assertThat(maybeAttribute).isNotPresent();
+        List<RtsStudentPackageAttribute> maybeAttribute = rtsService.findRtsStudentPackageAttributes("client", 1, new String[]{"attribute"});
```
Comment (jtreuting) [inline service/src/test/java/tds/student/services/impl/RtsServiceImplTest.java]: Change `maybeAttribute` to `attributes`

File: service/src/test/java/tds/student/web/endpoints/StudentControllerIntegrationTests.java [hunk 2]
```diff
@@ -65,22 +67,13 @@ public void shouldReturnNotFoundWhenStudentNotFound() throws Exception {
 
     @Test
     public void shouldReturnRtsAttributeForStudent() throws Exception {
-        when(rtsService.findRtsStudentPackageAttribute("client", 1, "testName")).thenReturn(Optional.of(new RtsStudentPackageAttribute("testName", "testValue")));
+        when(rtsService.findRtsStudentPackageAttributes(any(String.class), any(int.class), any(String[].class)))
+            .thenReturn(Collections.singletonList(new RtsStudentPackageAttribute("testName", "testValue")));
 
-        http.perform(get(new URI("/students/1/rts/client/testName"))
+        http.perform(get(new URI("/students/1/rts/client?attributeName=testName"))
```
Comment (jtreuting) [inline service/src/test/java/tds/student/web/endpoints/StudentControllerIntegrationTests.java]: Feels like we should test passing in multiple attributeNames as well as not providing them since it is a querystring param and therefore not required to route to that endpoint (as far as i know).  Not sure if it would provide `null` or empty array for attributeNames in the controller.

File: service/src/test/java/tds/student/services/impl/RtsServiceImplTest.java [hunk 3]
```diff
@@ -37,7 +38,7 @@ public void setUp() {
     public void shouldReturnEmptyIfPackageCannotBeFound() {
```
Comment (jtreuting) [inline service/src/test/java/tds/student/services/impl/RtsServiceImplTest.java:38]: Possibly add tests for passing in `null` or empty array (whatever the Controller would do when no querystring provided)

**Review by burzum619 — APPROVED** [review]

**Comment by millerts** [comment]
After a discussion with @jtreuting decided to do a new url structure.  Example below:
`http://localhost:8080/students/1/rts/client/attributes=test,test2`

Unfortunately, Spring Web integration tests don't seem to work with Matrix Variables.  Sigh.  I don't think it is critical path for now so I will follow back on this a little later.

Here is a good explanation of matrix variables: http://www.baeldung.com/spring-mvc-matrix-variables

File: service/src/test/java/tds/student/web/endpoints/StudentControllerIntegrationTests.java [hunk 4]
```diff
@@ -66,14 +70,29 @@ public void shouldReturnNotFoundWhenStudentNotFound() throws Exception {
...
 
     @Test
+    @Ignore("Spring MVC Integration tests does not currently handle Matrix Variables")
     public void shouldReturnRtsAttributeForStudent() throws Exception {
+        ArgumentCaptor<String[]> captor = ArgumentCaptor.forClass(String[].class);
         when(rtsService.findRtsStudentPackageAttributes(any(String.class), any(int.class), any(String[].class)))
             .thenReturn(Collections.singletonList(new RtsStudentPackageAttribute("testName", "testValue")));
 
-        http.perform(get(new URI("/students/1/rts/client?attributeName=testName"))
+        http.perform(get(new URI("/students/1/rts/client/attributes;names=testName,otherName"))
```
Comment (jtreuting) [inline service/src/test/java/tds/student/web/endpoints/StudentControllerIntegrationTests.java]: Is this the right URL?  Asking because in your comment on the PR you said it would look like `http://localhost:8080/students/1/rts/client/attributes=test,test2`

**Review by jtreuting — APPROVED** [review]
